release-21.1: kvcoord: make "txn unexpectedly committed" assertion opt-out

tbg opened the pull request:
Backport 1/1 commits from #73603.

/cc @cockroachdb/release

---

We are currently investigating how we are hitting this assertion,
but while it fires it is doing more harm than good.

See https://github.com/cockroachdb/cockroach/issues/67765.

The provided test exercises the case in which the error is returned
(i.e. opted out of the assertion).

Touches https://github.com/cockroachlabs/support/issues/1315.

Release note: None

tbg committed:
kvcoord: make "txn unexpectedly committed" assertion opt-out

We are currently investigating how we are hitting this assertion,
but while it fires it is doing more harm than good.

See https://github.com/cockroachdb/cockroach/issues/67765.

The provided test exercises the case in which the error is returned
(i.e. opted out of the assertion).

Touches https://github.com/cockroachlabs/support/issues/1315.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/73780)
<!-- Reviewable:end -->

tbg committed:
build: delete teamcity-bazel.sh

`release-21.1` doesn't use bazel, and removing this script will
remove the checks that make sure the bazel deps are up to date.

There seem to be issues with `make bazel-generate` (on this old branch)
and this is easier than trying to fix them.

Release note: None
